src/clients/HttpTunneler.cc

```diff
@@ -296,15 +296,23 @@ Http::Tunneler::handleResponse(const bool eof)
     }
 
     if (!parsedOk) {
+        debugs(11, 3, "Non-HTTP-compliant header:\n---------\n" << readBuf << "\n----------");
```

Comment:
This code suffers from the same two problems that equivalent HttpStateData::processReplyHeader() code suffers from. We can propose to duplicate those problems (for now), but (if we do) we should be explicit about it (so that it does not look as if we are unknowingly adding bugs):

```suggestion
        // XXX: This code and Server RESPONSE reporting code below duplicate
        // HttpStateData::processReplyHeader() reporting code, including its problems.
        debugs(11, 3, "Non-HTTP-compliant header:\n---------\n" << readBuf << "\n----------");
```

The above suggestion also documents code duplication itself.

Reply:
Done.